app/components/back-button.tsx

```diff
@@ -26,8 +25,14 @@ export const BackButton: FC<BackButtonProps> = ({ backUrl, hasFocus, ...props })
         minWidth: 'unset',
         padding: 0,
       }}
+      onClick={e => {
+        handleBack();
+        onClick?.(e);
+      }}
+      isDisabled={!hasBackState}
+      {...(props as any)}
     >
-      <ArrowIcon direction="left" color={backUrl === null || !hasFocus ? '#C1C3CC' : 'ink'} />
+      <ArrowIcon direction="left" color={!hasBackState || !isFocused ? '#c1c3cc' : 'ink'} />
```

Comment:
If the window is focused, but there is no back state, won't this show as 'ink', which we don't want?

Reply:
it is confusing to read, but works as expected because `!hasBackState` means there is no back state, and as such would be true, which would show `'#c1c3cc'` regardless of the status of isFocused

as verbose as possible

```tsx

const noBackState = !hasBackState;
const isNotFocused = !isFocused;
const isDimmed = noBackState || isNotFocused;
---
color={isDimmed ? '#c1c3cc' : 'ink'}
```

Reply:
```typescript
const hasDimmedColor = (hasBackAction: boolean, isFocused: boolean) => {
  return !hasBackAction || !isFocused;
};
```